src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -1080,21 +1080,18 @@ else if (needsPrepare)
      * @throws SQLServerException
      * @return the result set containing the meta data
      */
-    private ResultSet buildExecuteMetaData() throws SQLServerException {
+    private SQLServerResultSet buildExecuteMetaData() throws SQLServerException, SQLTimeoutException {
         String fmtSQL = userSQL;
 
-        ResultSet emptyResultSet = null;
+        SQLServerResultSet emptyResultSet = null;
         try {
             fmtSQL = replaceMarkerWithNull(fmtSQL);
             internalStmt = (SQLServerStatement) connection.createStatement();
             emptyResultSet = internalStmt.executeQueryInternal("set fmtonly on " + fmtSQL + "\nset fmtonly off");
-        } catch (SQLException sqle) {
+        } catch (SQLServerException sqle) {
```

Comment:
We're only catching a subset of exceptions possibly thrown from this class (SQLException -> SQLServerException) now, but we should also catch the other exceptions that could be thrown from here.

Reply:
The point of this PR is to _not_ catch exceptions.  All exceptions are now propagated, except the specific empty result set exception which is handled.  Previously all exceptions were caught, then a useless exception containing no details was thrown through `SQLServerException.makeFromDriverError`.  This PR does not change the circumstances under which exceptions are thrown, it simply ensures that the original exception is propagated with the useful details, instead of throwing a different exception with no information.  This is critical so that applications can handle errors appropriately.

Reply:
I see, thanks for the explanation.